File: src/main/java/mediathek/gui/actions/import_actions/ImportOldReplacementListAction.java
package mediathek.gui.actions.import_actions;

import javafx.application.Platform;
import javafx.scene.control.Alert;
import javafx.stage.FileChooser;
import mediathek.config.Konstanten;
import mediathek.tool.javafx.FXErrorDialog;

import javax.swing.*;
import java.awt.event.ActionEvent;

public class ImportOldReplacementListAction extends AbstractAction {
    public ImportOldReplacementListAction() {
        putValue(Action.NAME, "Alte Ersetzungstabelle...");
        putValue(Action.SHORT_DESCRIPTION, "Ermöglicht den Import der Ersetzungstabelle aus einer alten Konfigurationsdatei.");
    }

    @Override
    public void actionPerformed(ActionEvent e) {
        Platform.runLater(() -> {
            var fileChooser = new FileChooser();
            fileChooser.setTitle("Konfigurationsdatei öffnen");
            var selectedFile = fileChooser.showOpenDialog(null);
            if (selectedFile != null) {
                try {
                    var configReader = new OldConfigFileImporter();
                    var result = configReader.importAboBlacklist(selectedFile.getAbsolutePath(), false, false, true);
                    var alert = new ImportSettingsAlert(Alert.AlertType.INFORMATION);
                    String text = "Es wurden " + result.right + " Einträge importiert.";
                    alert.setContentText(text);
                    alert.showAndWait();
                } catch (Exception ex) {
                    Platform.runLater(() -> FXErrorDialog.showErrorDialog(Konstanten.PROGRAMMNAME,
                            "Fehler beim Importieren der Ersetzungstabelle",
                            "Es trat ein Fehler beim Import der Ersetzungstabelle auf.\nSollte dies häufiger auftreten kontaktieren Sie bitte das Entwicklerteam.",
                            ex));
                }
            } else {
                var alert = new ImportSettingsAlert(Alert.AlertType.WARNING);
                alert.setContentText("Der Import der Ersetzungstabelle wurde abgebrochen.");
                alert.showAndWait();
            }
        });
    }
}

File: src/main/kotlin/mediathek/gui/actions/import_actions/ImportOldReplacementListAction.kt
package mediathek.gui.actions.import_actions

import mediathek.config.Konstanten
import mediathek.mainwindow.MediathekGui
import mediathek.tool.FileDialogs.Companion.chooseLoadFileLocation
import mediathek.tool.SwingErrorDialog
import java.awt.event.ActionEvent
import javax.swing.AbstractAction
import javax.swing.JOptionPane

class ImportOldReplacementListAction : AbstractAction() {
    init {
        putValue(NAME, "Alte Ersetzungstabelle...")
        putValue(SHORT_DESCRIPTION, "Ermöglicht den Import der Ersetzungstabelle aus einer alten Konfigurationsdatei.")
    }

    override fun actionPerformed(e: ActionEvent) {
        val selectedFile = chooseLoadFileLocation(MediathekGui.ui(), " Konfigurationsdatei öffnen", "")
        if (selectedFile != null) {
            try {
                val configReader = OldConfigFileImporter()
                val (_, _, foundReplaceListEntries) = configReader.importAboBlacklist(selectedFile.absolutePath,
                                                                                      importAbo = false,
                                                                                      importBlacklist = false,
                                                                                      importReplaceList = true)
                val text = "Es wurden $foundReplaceListEntries Einträge importiert."
                JOptionPane.showMessageDialog(MediathekGui.ui(), text, Konstanten.PROGRAMMNAME, JOptionPane.INFORMATION_MESSAGE)
            }
            catch (ex: Exception) {
                val text = """
                    Es trat ein Fehler beim Import der Ersetzungstabelle auf.
                    Sollte dies häufiger auftreten kontaktieren Sie bitte das Entwicklerteam.
                    """.trimIndent()
                SwingErrorDialog.showExceptionMessage(MediathekGui.ui(), text, ex)
            }
        } else {
            JOptionPane.showMessageDialog(MediathekGui.ui(),
                                          "Der Import der Ersetzungstabelle wurde abgebrochen.",
                                          Konstanten.PROGRAMMNAME, JOptionPane.WARNING_MESSAGE)
        }
    }
}